skgstat/DirectionalVariogram.py

```diff
@@ -621,30 +660,10 @@ def _triangle(self, local_ref):
         DirectionalVariogram._circle
 
         """
-        # y coordinate is easy
-        c = self.bandwidth
-        gamma = np.radians(self.tolerance)
-        y = 0.5 * c
-
-        # a and h can be calculated
-        a = c / (2 * np.sin(gamma / 2))
-        h = np.sqrt((4 * a**2 - c**2) / 4)
-
-        # get the maximum x coordinate in the current representation
-        xmax = np.max(local_ref[:, 0])
-
-        # build the figure
-        poly = Polygon([
-            (0, 0),
-            (h, y),
-            (xmax, y),
-            (xmax, -y),
-            (h, -y)
-        ])
-
-        return poly
-
-    def _circle(self, local_ref):
+
+        raise NotImplementedError
+
```

Comment:
On the one hand, I like the idea of getting rid of the shapely dependency, on the other hand, we are removing the circle, triangle and custom shaped search areas. At least the circle and triangle area shapes are published methods and I am wondering if we should really drop their support for the sake of calculation speed.
What do you think of keeping the old methods and add your code as a new *model*, called 'polar' or something?
It would be the default method and shapely could be an optional dependency. So if `directional_model` is not `'polar'`(or whatever it's called then), the class can check if shapely is installed and raise an error if not.

This way, `DirectionalVariogram` would be significantly faster, but the user can decide to be more flexible at the cost of speed, id needed.

Reply:
I think we could implement these two using the same methodology as the compass one - the math just gets a bit trickier...

For example, to constrain the angles to a rectangle around the azimith of badwidth width:

    lengh <= bandwidth / sin(angle - azimuth)

to create the triangle directional model, we'd need to combine this with something similar to what the compass one does,:

    (|angle - azimuth| <= tolerance) & (lengh <= bandwidth / sin(angle - azimuth))

Reply:
I just wanted to implement the simple one first and see what you thought about it before I spent the time to implement all of them.

Oh, and I don't know how complicated the math gets for the circle on. Hm...

Reply:
Just pushed a change to implement the triangle model and set that as default again. Wasn't too hard :)